### PR TITLE
Directly calling response

### PR DESCRIPTION
**Pull request by fabiob**
I've added some tests for the documented way to write to the response directly, as in: https://github.com/pleerock/routing-controllers#using-request-and-response-objects. It looks like #241 broke this usage pattern.

### CODE REVIEW

**Comment by fabiob** [comment]
I've included a proposal to detect whether the user code handled the response manually or not.

**Comment by fabiob** [comment]
P.S.: the failing tests are newly detected failures from the `assertRequest` change I've made.

File: src/driver/koa/KoaDriver.ts [hunk 1]
```diff
@@ -209,6 +209,12 @@ export class KoaDriver extends BaseDriver implements Driver {
      */
     handleSuccess(result: any, action: ActionMetadata, options: Action): void {
 
+        // if the action returned the context or the response object itself, short-circuits
+        if (result && (result === options.response || result === options.context)) {
+            options.next();
```
[the remark this answers is not in the document]
Comment (fabiob) [reply src/driver/koa/KoaDriver.ts]: Thanks. Fixed it.

**Comment by NoNameProvided** [comment]
@19majkel94 we can merge right? or do we want to wait for a review from @pleerock ?

**Comment by 19majkel94** [comment]
I would wait for @pleerock review due to changes in test utils.

**Comment by fabiob** [comment]
Rebased to include latest `master`.

**Comment by fabiob** [comment]
Would it help if I extract the test utils into another PR?

**Comment by fabiob** [comment]
OK, I've extracted the test-utils change into another PR (#297).

@NoNameProvided @19majkel94, do you guys think we can merge this faster?

**Comment by 19majkel94** [comment]
> @NoNameProvided @19majkel94, do you guys think we can merge this faster?

Yes, but we still have to wait for @pleerock to release the merged PRs on npm 😜

**Comment by NoNameProvided** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**